File: Data Structures & Algorithms/maximum-subarray/submission-0.js
class Solution {
  /**
   * @param {number[]} nums
   * @return {number}
   */
  maxSubArray(nums) {
    let maxSub = nums[0];
    let currSum = 0;

    for (const num of nums) {
      if (currSum < 0) currSum = 0;
      currSum += num;
      maxSub = Math.max(maxSub, currSum);
    }
    return maxSub;
  }
}

File: Data Structures & Algorithms/duplicate-integer/submission-0.cpp
class Solution {
public:
    bool hasDuplicate(std::vector<int>& nums) {

        std::sort(nums.begin(),nums.end());
        const int SIZE = nums.size();
        for(int i = 1; i < SIZE;i++){

            if(nums[i] == nums[i-1])
                return true;

        }

        return false;

    }
};
